website_snippet_carousel_product/static/src/js/s_carousel_product_frontend.js

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Tecnativa - Alexandre Díaz
+// License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+odoo.define("website_snippet_carousel_product.s_carousel_product", function (require) {
+    "use strict";
+
+    var core = require("web.core");
+    var sAnimation = require("website.content.snippets.animation");
+
+    var _t = core._t;
+
+    sAnimation.registry.js_carousel_product = sAnimation.Class.extend({
+        selector: ".js_carousel_product",
+
+        _refreshAnimations: function () {
+            this.trigger_up("animation_start_demand", {
+                editableMode: false,
+                $target: this.$target.find(".product_carousel_slider"),
+            });
+        },
+
+        /**
+         * @override
+         */
+        start: function () {
+            var self = this;
+            var limit = Number(this.$target.attr("data-products-limit")) || 12;
+            var domain = this.$target.attr("data-domain") || false;
+            var products_per_slide =
+                Number(this.$target.attr("data-products-per-slide")) || 4;
+            var template =
+                this.$target.attr("data-template") ||
+                "website_snippet_carousel_product.s_carousel_product_items";
+
+            // Compatibility with db that saved content inside by mistake
+            this.$target.empty();
+            // Prevent user edition
+            this.$target.attr("contenteditable", "False");
+
+            var def = $.Deferred();
```

Comment:
Aren't we supposed to use promises in v12? :thinking: 

```suggestion
            var def = new Promise();
```